app_feup/lib/controller/local_storage/app_shared_preferences.dart

```diff
@@ -71,8 +74,25 @@ class AppSharedPreferences {
     prefs.remove(userPw);
   }
 
+  /// Returns a tuple containing the user's student number and password
+  /// and faculties he is associated to.
+  ///
+  /// *Note:*
+  /// * the first element in the tuple is the user's student number.
+  /// * the second element in the tuple is the user's password, in plain text
+  ///   format.
+  /// * the third element in the tuple is a list of strings
+  ///   with the code of the faculty.
+  static Future<Tuple3<String, String, List<String>>>
```

Comment:
A tuple of 3 values is already starting to sound a bit too complex to maintain. Can you create a model for this instead?

Reply:
That may be a good idea if we're looking into the future and seeing this combination of values being required often. Is that what you had in mind? Because if it is rather used very sparingly or only in this situation it might be a better option to dump the function all together and use two separate ones instead